src/api/centra/utils/formatProduct.js

```diff
@@ -25,7 +24,8 @@ export default function formatProduct(inProduct, inFormatters, filterProperties)
     product[optionKey] = typeof optionValue === 'function' ? optionValue(product) : optionValue
   })
 
-  return product
+  // filter out properties using filterProperties
+  return filterProperties ? filterProduct(product, filterProperties) : product
```

Comment:
Looks like this now requires the user to define their formatters twice, once in the formatters object and once in the filter array. Let's think about how we can solve this use case in a better way.